### PR TITLE
Update bazel deps

### PR DESCRIPTION
**Pull request by ixdy**
**What this PR does / why we need it**: updates relevant bazel dependencies to pull in minor feature enhancements and bug fixes:
* rules_go 0.15.3: some speed improvements and better compatibility on Windows and Mac, among other bugfixes / enhancements
* repo-infra: fixes needed to support rules_go 0.15.1, and also a fix for https://github.com/kubernetes/kubernetes/pull/65501#issuecomment-400761696
* rules_docker v0.5.1: various enhancements/fixes, but we need this to support new manifest lists from upstream
* busybox latest: we last updated this a while ago. since Docker hub now uses manifest lists for core images like this, we needed the rules_docker update
* gazelle 0.14.0: among other enhancements, adds a feature to remove the need to build protoc for every build, which we aren't even using.

**Special notes for your reviewer**:
Depends on https://github.com/kubernetes/repo-infra/pull/81. I believe we also need bazel 0.16+.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: build/root/BUILD.root [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 # gazelle:exclude _gopath
 # gazelle:exclude _output
 # gazelle:exclude _tmp
+
 # gazelle:prefix k8s.io/kubernetes
 
 # Disable proto rules, since the Go sources are currently generated by
 # hack/update-generated-protobuf.sh and checked into the repo.
-# gazelle:proto disable
+# gazelle:proto disable_global
```
Comment (ixdy) [inline build/root/BUILD.root]: > disable_global: like disable mode, but also prevents Gazelle from using any special cases in dependency resolution for Well Known Types and Google APIs. Useful for avoiding build-time dependencies on protoc.

File: hack/update-bazel.sh [hunk 2]
```diff
@@ -41,7 +41,6 @@ if ! grep -q "# gazelle:importmap_prefix" "${KUBE_ROOT}/vendor/BUILD"; then
 fi
 
 gazelle fix \
-    -build_file_name=BUILD,BUILD.bazel \
```
Comment (ixdy) [inline hack/update-bazel.sh]: this moved to the root BUILD.bazel file

**Comment by ixdy** [comment]
The build failure appears to be https://github.com/bazelbuild/rules_go/issues/1592. We might be stuck until that gets fixed.

**Comment by ixdy** [comment]
Easiest way to reproduce the failure on this PR:
```
bazel test --features=race //pkg/master:go_default_test
```

**Comment by ixdy** [comment]
ah, right. this only fails on targets where we explicitly turn race detection off in the `go_test` rule.

**Comment by BenTheElder** [comment]
/hold
(to ward off @fejta-bot until https://github.com/bazelbuild/rules_go/issues/1592 is sorted)

**Comment by ixdy** [comment]
Updated to rules_go 0.15.3. This should build and test successfully now.

**Comment by spiffxp** [comment]
/lgtm
/milestone v1.12
/priority critical-urgent
SIG Testing chair, I do not expect this to impact release-blocking builds, since those don't use bazel

**Pull request merged**